compiler-plugin-test/src/test/java/io/ballerina/stdlib/persist/compiler/plugin/CompilerPluginTest.java

```diff
@@ -183,6 +183,69 @@ public void testTableName() {
                 DiagnosticsCodes.PERSIST_113.getCode(), 1);
     }
 
+    @Test
+    public void testInvalidInitialisation() {
+        DiagnosticResult diagnosticResult = loadPackage("package_14").getCompilation().diagnosticResult();
+        List<Diagnostic> errorDiagnosticsList = diagnosticResult.diagnostics().stream().
+                filter(r -> r.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR)).
+                collect(Collectors.toList());
+        assertValues(errorDiagnosticsList,
+                "invalid entity initialisation: the associated entity of this[Item] does not have " +
+                        "the field with the relationship type", DiagnosticsCodes.PERSIST_115.getCode(), 2);
+    }
+
+    @Test
+    public void testInvalidInitialisation1() {
+        DiagnosticResult diagnosticResult = loadPackage("package_15").getCompilation().diagnosticResult();
+        List<Diagnostic> errorDiagnosticsList = diagnosticResult.diagnostics().stream().
+                filter(r -> r.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR)).
+                collect(Collectors.toList());
+        assertValues(errorDiagnosticsList, "invalid entity initialisation: the @Relation annotation cannot be " +
+                        "added to both entities, if relationship of these entities is 1:1 or 1:M",
```

Comment:
Shall we say, The relation annotation should only be added to the relationship owner for one-to-one and one-to-many associations.